### ISSUE TITLE
Database dropdown selection has no effect for Athena

### ISSUE THREAD

**Issue by dangeReis** [dbeaver/dbeaver#11196]
#### System information: 
- Operating system (distribution) and version: Windows 10
- DBeaver version: 7.3.4.202101311042
- Additional extensions: none

#### Connection specification:
- Database name and version: Athena
- Driver name: Simba
- Do you use tunnels or proxies (SSH, SOCKS, etc)? no

#### Describe the problem you're observing:
The selected database does not get set.  I think the fix would be setting the Schema parameter in the driver.  It is currently possible to set the parameter, so that it no longer looks in the default db, but would be useful if the drop down actually worked.  

#### Steps to reproduce, if exist:
Connect to athena.  Try to select from a table not in the "default" database.  Get an error.

`
#### Include any warning/errors/backtraces from the logs

`SQL Error [100071] [HY000]: [Simba][AthenaJDBC](100071) An error has been thrown from the AWS Athena client. SYNTAX_ERROR: line 1:15: Table awsdatacatalog.default.tablename does not exist

**Comment by chanansh** [comment]
Hi @uslss, the current version is 7.3.4 what does it mean this was added to 21.0.0?
Moreover, any workarounds? Even if I set the schema as default in the connection settings it is still ignored.

**Comment by dangeReis** [comment]
> Even if I set the schema as default in the connection settings it is still ignored.

This part actually works for me.  The workaround is to set up separate connections per database  (or use fully qualified db names)

**Comment by flambchap** [comment]
Hi @chanansh. We are switching to a [new versioning scheme](https://github.com/dbeaver/dbeaver/discussions/11024). The 21.0.0 version will be out on the 1st of March.

**Comment by ShadelessFox** [comment]
Hello @dangeReis,

Can you please provide more details about your problem?

Are you talking about dialog for changing active catalog/schema (<kbd>Ctrl+0</kbd>)?

**Comment by dangeReis** [comment]
> Hello @dangeReis,
> 
> Can you please provide more details about your problem?
> 
> Are you talking about dialog for changing active catalog/schema (Ctrl+0)?

Yes, that is the part that is not working.  I can pick a schema, but the query still runs against the wrong schema.

**Comment by ShadelessFox** [comment]
I see, thanks for the explanation.

**Comment by ShadelessFox** [comment]
Looks like the issue is with the driver itself. Need further examination.

**Comment by dangeReis** [comment]
> Looks like the issue is with the driver itself. Need further examination.

Possible, but setting a default in settings works, so the driver is at least is capable of accepting default db.  I'm happy to try a different driver if you think it will help.

**Comment by ShadelessFox** [comment]
It may be capable of setting default schema during connection initialization, but may not be able to change it later for active connection.

Unfortunately, I can't name any alternatives. But, at least for now, you can use the workaround mentioned earlier.

**Comment by dangeReis** [comment]
> It may be capable of setting default schema during connection initialization, but may not be able to change it later for active connection.
> 
> Unfortunately, I can't name any alternatives. But, at least for now, you can use the workaround mentioned earlier.

I meant like a newer version of the driver, if the one I'm using is having issues.  Please test and let me know if this is indeed an issue with the driver and if updating to a new version resolves it.

**Comment by ShadelessFox** [comment]
Updating to newer version (2.0.21.1000) does not resolve the issue.

**Comment by dangeReis** [comment]
> Updating to newer version (2.0.21.1000) does not resolve the issue.

So are we sure the issue is with the driver?  In that case can we open an issue with them?

**Comment by dangeReis** [comment]
I do see this note on the documentation page:

`To use the JDBC driver for multiple data catalogs with Athena (for example, when using an external Hive metastore or federated queries), include MetadataRetrievalMethod=ProxyAPI in your JDBC connection string.`

Not sure if it's relevant here.  


https://docs.aws.amazon.com/athena/latest/ug/connect-with-jdbc.html

**Comment by ShadelessFox** [comment]
I can't find any other traces of this problem, so that's my guess.

Unfortunately, that note is not relevant here, since we're talking about changing current schema, not the catalog.

As I said, we need further examination. I will write back later.

**Comment by ShadelessFox** [comment]
We have concluded that the issue is indeed with the driver. You can create a separate issue on their board.

**Comment by dangeReis** [comment]
> We have concluded that the issue is indeed with the driver. You can create a separate issue on their board.

I can try.  Do you have some sample code that reproduces the issue with the driver?

**Comment by ShadelessFox** [comment]
As a starting point, the following code sample can be used:
https://simba.wpengine.com/products/Athena/doc/JDBC_InstallGuide/content/jdbc/ath/examples/jdkexamples.htm

And block to reproduce the issue:
```java
conn.setSchema("non-default-schema");
conn.getSchema(); // will return "non-default-schema"

try (Statement stmt = conn.createStatement()) {
    try (ResultSet rs = stmt.executeQuery("select * from some_table_from_non_default_schema")) {
        rs.next();
    }
}
